proposals/Reliable Mutability Primitive.md

```diff
@@ -0,0 +1,166 @@
+- Fix IPNS publishing in browsers.
+- Enable pubsub by default + IPNS over pubsub.
+- Implement & deploy strategic republishing system on PL hosted nodes that will republish popular content and drop stalled one. So that IPNS works even thought pinning services.
+- (Nice to have) Implement and rollout delegated publishing to enable updating from multiple devices.
+- (Nice to have) Implement and rollout versioning system so that nodes can see full update history.
+
+##### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+Publishing IPNS record takes just a few miliseconds. Shutting laptop and loading `https://ipfs.io/ipns/{key}` week later on other device loads the content published as fast as loading corresponding `https://ipfs.io/ipfs/{cid}`.
+
+
+#####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+People and teams are storing IPNS keys in [DNSLink][] records as opposed to CIDs, because that provides a more effective and simpler way to publish updates. We see new projects are leveraging IPNS over working around it via things like [DNSLink][].
+
+
+##### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+- IPNS has bad reputation, changing it may prove to be a challenge.
+- Teams may find that general primitive is not adequate for custom needs.
+- Required key management (e.g. key recovery, key compromise) still may prove it impractical.
+- Higher level alternatives like textile threads may prove to be better layer of abstraction.
+
+##### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+- Integrating textile threads as core component of the ecosystem.
+- Standardizing DNS record updates mechanism and intergating into IPFS.
+- Work with Pinning Services to implement stardized DNSLink updating mechanism.
+
+##### Dependencies/prerequisites
+<!--List any other projects that are dependencies/prerequisites for this project that is being pitched.-->
+
+- Functioning IPNS in browsers
+- Enabling pubsub in go-ipfs
+
+##### Future opportunities
+<!--What future projects/opportunities could this project enable?-->
+
+- [Petname][] system for memorable but non canonical names (kind of how our phones allow us to not remember numbers).
+- Subscribtion based deny lists (node can subscribe to any endpoint)
+- RSS like systems (In RSS readers memorable namse were not a concern)
+- DAGs revisions carrying IPNS key so that last latest revision or any past revision could be discovered.
+
```

Comment:
Also, more advanced mutability systems could be built (e.g. where instead of a single public key, we can have M of N signers required on records). But it all starts from having the UX and process around pinning mutable names and having a performant and efficient mutability system (e.g. Persistent PubSub, like what IPNS over PubSub uses)